### PR TITLE
Bug 3098

### PR DESCRIPTION
**Pull request by stgallagher**
Added structure necessary to edit the 'orig_due_date' in the PPG_details records of a participant, through the participant's Status History tab view.

### CODE REVIEW

File: app/controllers/ppg_details_controller.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  def update
+    @participant = Participant.find(params[:participant_id])
+    @person = @participant.person
+    @ppg_detail = PpgDetail.find(params[:id])
+
+    respond_to do |format|
+      if @ppg_detail.update_attributes(params[:ppg_detail])
+        flash[:notice] = 'Original Due Date was successfully updated.'
+        format.html { redirect_to(participant_path(@participant.id)) }
+        format.json  { render :json => @ppg_details }
```
Comment (hannahwhy) [inline app/controllers/ppg_details_controller.rb]: Why do you need to provide a JSON response?

File: app/views/ppg_details/edit.html.haml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+- lbl = "Edit Due Date"
+- lbl += " for #{@person}" if @person
```
Comment (hannahwhy) [inline app/views/ppg_details/edit.html.haml:2]: I'd prefer a helper method that formats a `Person` for this case, or explicit usage of `Person#full_name`. 

(I'd also like to see `Person#to_s` go away; it throws away too much information.)

File: app/views/participants/_details.html.haml [hunk 3]
```diff
@@ -4,6 +4,7 @@
       %b
         Due Date:
       = blank_safe(@participant.due_date, "Unknown")
+      = link_to "Edit Due Date", edit_participant_ppg_detail_path(@participant.id, @participant.ppg_details.first.id), :class => "edit_link icon_link"
```
Comment (hannahwhy) [inline app/views/participants/_details.html.haml:7]: Why only the most recent `PpgDetail`?

File: app/views/ppg_details/edit.html.haml (hunk 2 above)
Comment (stgallagher) [inline app/views/ppg_details/edit.html.haml:2]: Added [ticket](https://code.bioinformatics.northwestern.edu/issues/issues/show/3112)

File: app/views/participants/_details.html.haml (hunk 3 above)
Comment (stgallagher) [inline app/views/participants/_details.html.haml:7]: The Participant should only have a single PpgDetail record. Wrote up [this](https://code.bioinformatics.northwestern.edu/issues/issues/show/3111) ticket.

File: app/controllers/ppg_details_controller.rb (hunk 1 above)
Comment (stgallagher) [inline app/controllers/ppg_details_controller.rb]: It's not needed. Deleted.

**Comment by stgallagher** [comment]
Pull Request Ready for Review.

File: app/controllers/ppg_details_controller.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
+class PpgDetailsController < ApplicationController
+
+  def edit
+    @participant = Participant.find(params[:participant_id])
+    @person = @participant.person
+    @ppg_detail = PpgDetail.find(params[:id])
+
+    respond_to do |format|
+      format.html
+    end
```
Comment (hannahwhy) [inline app/controllers/ppg_details_controller.rb]: If the action provides only a single response type, you don't need the `respond_to` / `format` machinery.

**Comment by stgallagher** [comment]
Pull Request Ready for Review.

File: app/controllers/ppg_details_controller.rb (hunk 4 above)
Comment (stgallagher) [inline app/controllers/ppg_details_controller.rb]: Fixed.

**Pull request merged**